gcalcli/validators.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import absolute_import
+
+import re
+
+from six.moves import input
+
+from gcalcli.utils import valid_override_colors, REMINDER_REGEX
+from gcalcli.exceptions import ValidationError
+
+from dateutil.parser import parse
+
+
+def get_input(printer, prompt, validator_func):
+    printer.msg(prompt, 'magenta')
+    while True:
+        try:
+            output = validate_input(validator_func)
+            return output
+        except ValidationError as e:
+            printer.msg(e.message, "red")
+            printer.msg(prompt, 'magenta')
+
+
+def valid_colors_validator(input_str):
+    """
+    A filter allowing only the following strings:
+        * basil
+        * banana
+        * tomato
+        * lavender
+        * flamingo
+        * blueberry
+        * peacock
+        * grape
+    Raises ValidationError otherwise.
+    """
+    valid_override_colors.append("")  # allow empty
+    try:
+        assert input_str in valid_override_colors
+        return input_str
+    except AssertionError:
+        raise ValidationError("Expected colors are: basil, peacock, grape, "
+                              "lavender, blueberry, tomato, flamingo or "
+                              "banana. (Ctrl-C to exit)\n")
+
+
+def str_to_int_validator(input_str):
+    """
+    A filter allowing any string which can be
+    converted to an int.
+    Raises ValidationError otherwise.
+    """
+    try:
+        int(input_str)
+        return input_str
+    except ValueError:
+        raise ValidationError(
+            "Input here must be a number. (Ctrl-C to exit)\n"
+        )
+
+
+def parsable_date_validator(input_str):
+    """
+    A filter allowing any string which can be parsed
+    by dateutil.
+    Raises ValidationError otherwise.
+    """
+    try:
+        parse(input_str)
```

Comment:
Here we actually need to use the `utils.get_time_from_str` function as the parse function.  That function does this handy-dandy fuzzy-matching stuff which is very convenient.  You can say stuff like "tomorrow 3pm" instead of being very formal with the input formats.  I know that I rely on this functionality because I can never remember exactly which format to provide otherwise :stuck_out_tongue_winking_eye: 

My apologies that I didn't catch this on the first pass.

Reply:
I think `dateutils.parse()` does _most_ of the handy-dandy fuzzy-matching stuff, but you're right it doesn't recognise "tomorrow", which I rely on too. I love handy-dandy date matchers! I will look at this in a couple hours and submit the patch.

Reply:
Awesome.  I am very happy with this work, it fits in very well with my mission to provide better modularity to the code.  So thank you once again.

Reply:
Making that change now...

Reply:
Have pushed those changes now. Hope they do the job. I hadn't spotted that very handy function. Great to be finding out more about this code all the time. Thank you for your patience unpicking my work - I'm learning so much. :+1: